File: LASYA-PPS/PPS-RefuteTesting/CaesarCipher.py
def encrypt(text, shift):
    result = ""
    for char in text:
        if char == ' ':
            raise ValueError("Spaces are not allowed in the input text.")
        elif char.isalpha():  # Check if the character is a letter
            shift_base = ord('A') if char.isupper() else ord('a')
            result += chr((ord(char) - shift_base + shift) % 26 + shift_base)
        else:
            # Encrypt special characters using their ASCII values
            result += chr((ord(char) + shift) % 128)
    return result

# Test the function
try:
    text = input("Enter the text to encrypt: ")
    shift = int(input("Enter the shift value: "))
    encrypted_text = encrypt(text, shift)
    print(f"Encrypted text: {encrypted_text}")
except ValueError as e:
    print(e)

File: LASYA-PPS/PPS-RefuteTesting/CaesarCipher.c
#include <stdio.h>
#include <ctype.h>
#include <string.h>

// Function to encrypt the text
void encrypt(char text[], int shift, char result[]) {
    int i;
    for (i = 0; text[i] != '\0'; i++) {
        char char_result;
        if (isalpha(text[i])) {  // Check if the character is a letter
            char shift_base = isupper(text[i]) ? 'A' : 'a';
            char_result = (text[i] - shift_base + shift) % 26 + shift_base;
        } else {
            char_result = text[i];  // Non-alphabet characters are not encrypted
        }
        result[i] = char_result;
    }
    result[i] = '\0';  // Null-terminate the result string
}

int main() {
    char text[1000];
    int shift;
    char encrypted_text[1000];

    // Get user input
    printf("Enter the text to encrypt: ");
    fgets(text, sizeof(text), stdin);
    text[strcspn(text, "\n")] = '\0';  // Remove the newline character

    printf("Enter the shift value: ");
    scanf("%d", &shift);

    // Encrypt the text
    encrypt(text, shift, encrypted_text);

    // Display the encrypted text
    printf("Encrypted text: %s\n", encrypted_text);

    return 0;
}
